### PR TITLE
V3 Migration Guide

### PR DESCRIPTION
**Pull request by arschles**
Intended to be a guide for developers to do a self-serve migration. Assumes that auth migration is complete.

@rdallman @nikhilm @nicksardo @edsrzf @romand PTAL

### CODE REVIEW

File: mq/v2_v3_migration_guide/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+If you're using an [officially supported client library](/mq/3/libraries/), simply change
+your dependency to use the `v3` branch. If you're confused about how to do this, please
+[contact our support team](/support/).
+
+If you're using a third party library, you may need to alter it to reflect the new IronMQ v3
+API. See the [IronMQ v3 changelog](/mq/3/reference/api/#changes) for an
+overview of what's changed and how you should alter the client to work with v3.
+
+# Step 3: Audit Your app
```
Comment (nikhilm) [inline mq/v2_v3_migration_guide/index.md]: Nit app -> App

File: mq/v2_v3_migration_guide/index.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+title: IronMQ v3 Migration Guide
+summary: "The complete guide to moving your app from IronMQ v2 to IronMQ v3"
```
Comment (nikhilm) [inline mq/v2_v3_migration_guide/index.md]: complete sounds a little pompous considering it's 50 lines :) Just my opinion but I'd remove it.

**Comment by nikhilm** [comment]
LGTM with comments fixed.

File: mq/v2_v3_migration_guide/index.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- v3's introduces a `reserve` call for clients to use. This call replaces v2's `get` call.
+- v3's `get` is only intended to view a message, not for processing it
+- `reserve` returns a `reservation_id`, which must be used in later calls to [delete](/mq/3/reference/api/#delete-messages) or [release](/mq/3/reference/api/#release-message) the message, or [extend its reservation](/mq/3/reference/api/#touch-message)
+
+See [IronMQ v3 Basics](/mq/3/reference/api/#changes) for more details on these new concepts.
+
+# Step 2: Update Your Library
+
+If you're using an [officially supported client library](/mq/3/libraries/), simply change
+your dependency to use the `v3` branch. If you're confused about how to do this, please
```
Comment (rdallman) [inline mq/v2_v3_migration_guide/index.md:32]: imo we should switch v3 branch to master and make 'v2' branches (this is true at this point, just raising it -- no need to change here)

**Comment by rdallman** [comment]
thanks

**Pull request merged**